packages/core/src/lib/lit-core.ts

```diff
@@ -330,39 +328,55 @@ export class LitCore {
           const delta: string[] = validatorData.bootstrapUrls.filter((item) =>
             existingNodeUrls.includes(item)
           );
-          // if the sets differ we reconnect.
+
+          // check if the node sets are non-matching and re-connect if they do not.
           if (delta.length > 1) {
-            // check if the node sets are non-matching and re-connect if they do not.
             /*
-                TODO: This covers *most* cases where a node may come in or out of the active
-                set which we will need to re attest to the execution environments.
-                However, the sdk currently does not know if there is an active network operation pending.
-                Such that the state when the request was sent will now mutate when the response is sent back.
-                The sdk should be able to understand its current execution environment and wait on an active
-                network request to the previous epoch's node set before changing over.
-              */
+                  TODO: This covers *most* cases where a node may come in or out of the active
+                  set which we will need to re attest to the execution environments.
+                  However, the sdk currently does not know if there is an active network operation pending.
+                  Such that the state when the request was sent will now mutate when the response is sent back.
+                  The sdk should be able to understand its current execution environment and wait on an active
+                  network request to the previous epoch's node set before changing over.
+                */
             log(
               'Active validator sets changed, new validators ',
               delta,
               'starting node connection'
             );
+            await this.connect(); // Will update `epochInfo`
           }
-
-          await this.connect();
-        } catch (err: unknown) {
-          // FIXME: We should emit an error event so that consumers know that we are de-synced and can connect() again
-          // But for now, our every-30-second network sync will fix things in at most 30s from now.
-          // this.ready = false; Should we assume core is invalid if we encountered errors refreshing from an epoch change?
-          const { message = '' } = err as
-            | Error
-            | NodeClientErrorV0
-            | NodeClientErrorV1;
-          logError(
-            'Error while attempting to reconnect to nodes after epoch transition:',
-            message
+        } else {
+          // In case of centralised networks, we don't run `connect()` flow, so we will manually update epochInfo here
```

Comment:
Why is a centralized network treated differently? The only difference should be in the attestation but the handshake should be invariant to the network?

Reply:
My understanding is that for centralized networks, the node list doesn't change, so re-handshaking with all the nodes on epoch change would be redundant; if that's not true, we should definitely fix it...